packages/cozy-harvest-lib/src/datacards/GeoDataCard.jsx

```diff
@@ -0,0 +1,229 @@
+      for (let feature of tripsRef.current) {
+        mapRef.current.removeLayer(feature)
+      }
+      mapRef.current.fitBounds(tripsRef.current[index].getBounds())
+      mapRef.current.addLayer(tripsRef.current[index])
+    }
+  }, [index])
+  return <div style={mapStyle} ref={nodeRef} />
+}
+
+const useCarrousel = (initialState, min, max) => {
+  const [curIndex, setIndex] = useState(0)
+
+  const handlePrev = useCallback(() => {
+    const prev = curIndex - 1
+    setIndex(prev < min ? max : prev)
+  }, [curIndex, min, max])
+
+  const handleNext = useCallback(() => {
+    const next = curIndex + 1
+    setIndex(next > max ? min : next)
+  }, [curIndex, max, min])
+
+  return [curIndex, handlePrev, handleNext]
+}
+
+const TripInfoSlideRaw = ({ trip }) => {
+  return (
+    <div>
+      <Media>
+        <Img>
+          <Icon icon={FlagIcon} color="var(--emerald)" className="u-mr-half" />
+        </Img>
+        <Bd>
+          <Typography variant="body1">
+            {getStartPlaceDisplayName(trip)}
+          </Typography>
+          <Typography variant="caption">
+            {getStartPlaceCaption(trip)}
+          </Typography>
+        </Bd>
+      </Media>
+      <Media>
+        <Img>
+          <Icon
+            icon={FlagIcon}
+            color="var(--pomegranate)"
+            className="u-mr-half"
+          />
+        </Img>
+        <Bd>
+          <Typography variant="body1">
+            {getEndPlaceDisplayName(trip)}
+          </Typography>
+          <Typography variant="caption">{getEndPlaceCaption(trip)}</Typography>
+        </Bd>
+      </Media>
+    </div>
+  )
+}
+
+const TripInfoSlide = memo(TripInfoSlideRaw)
+
+const GeoDataCard = ({ trips }) => {
+  const enhancedTrips = useMemo(() => prepareTrips(trips), [trips])
+  const [index, setPrev, setNext] = useCarrousel(0, 0, trips.length - 1)
+  return (
+    <Card className="u-ph-0 u-pb-0">
+      <Media className="u-mb-1">
+        <Img className="u-pl-half">
+          <IconButton onClick={setPrev}>
+            <Icon icon={LeftIcon} />
+          </IconButton>
+        </Img>
+        <Bd>
+          <Box ml={1} mr={1}>
+            <SwipeableViews index={index} disabled>
+              {enhancedTrips.map(trip => {
+                return <TripInfoSlide key={trip.id} trip={trip} />
+              })}
+            </SwipeableViews>
+          </Box>
+        </Bd>
+        <Img className="u-pr-half">
+          <IconButton onClick={setNext}>
+            <Icon icon={RightIcon} />
+          </IconButton>
+        </Img>
+      </Media>
+      <TripsMap trips={enhancedTrips} index={index} />
+    </Card>
+  )
+}
+
+const makeQueryFromProps = ({ accountId }) => ({
+  query: Q('io.cozy.timeseries.geojson')
+    .where({
+      'cozyMetadata.sourceAccount': accountId
+    })
+    .sortBy([
+      { 'cozyMetadata.sourceAccount': 'desc' },
+      { 'cozyMetadata.createdAt': 'desc' }
+    ])
+    .indexFields(['cozyMetadata.sourceAccount', 'cozyMetadata.createdAt'])
```

Comment:
I think [startDate]( https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.timeseries.md#iocozytimeseriesgeojson) is more reliable than `cozyMetadata.createdAt` as it specifies the actual start of the time series.

Reply:
I think in the context of a konnector that the notion of recency should be mapped to the notion of import, this is why I used createdAt. And, this will enable consistency across several doctypes. As this is fairly easy to change, I propose to keep this as is and ask about this after a demo or after more usage.